ggshield/core/client.py

```diff
@@ -20,7 +20,16 @@ def create_client_from_config(config: Config) -> GGClient:
             # This can happen when the user first tries the app and has not gone through
             # the authentication procedure yet. In this case, replace the error message
             # complaining about an unknown instance with a more user-friendly one.
-            raise APIKeyCheckError(e.instance, "GitGuardian API key is needed.")
+            raise APIKeyCheckError(
+                e.instance,
+                "A GitGuardian API key is needed to use ggshield.\n"
+                "To get one, authenticate to your dashboard by running :\n\n"
+                "\t`ggshield auth login`\n\n"
+                "If you are using an on-prem version of GitGuardian, "
+                "use the --instance option to point to it.\n"
+                "Read the following documentation for more information : "
```

Comment:
```suggestion
                "To get one, authenticate to your dashboard by running:\n\n"
                "\t`ggshield auth login`\n\n"
                "If you are using an on-prem version of GitGuardian, "
                "use the --instance option to point to it.\n"
                "Read the following documentation for more information: "
```

No spaces before `:` in English :)

Also, maybe you can use a triple-quote string for the message?

Reply:
Fixed 👌   
About the triple-quote, I'm not sure it is more readable... we would have something like that :

```
            raise APIKeyCheckError(
                e.instance,
                """A GitGuardian API key is needed to use ggshield.
To get one, authenticate to your dashboard by running:

    `ggshield auth login`

If you are using an on-prem version of GitGuardian, \
use the --instance option to point to it.
Read the following documentation for more information: \
https://docs.gitguardian.com/ggshield-docs/reference/auth/login
""",
            )
```

Let me know if you still want to go with the triple quotes, I'll do the changes.

Reply:
I think it's easier to imagine the end result: for example I didn't catch that there were back-ticks around `ggshield auth login`, which I would remove: the surrounding blank lines and the indentation are enough to understand it's a command, and not having the back-ticks makes it easier to copy-and-paste.

Reply:
Makes sense.  
Here is how it looks : 
<img width="1087" alt="image" src="https://user-images.githubusercontent.com/24247716/225089744-164f3394-f255-47b5-ba43-e560d2f7eaa1.png">